AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -60,7 +70,34 @@ - (BOOL)canLayoutAsynchronous
 
 - (id<ASLayoutElement>)finalLayoutElement
 {
-  return self;
+  if (ASLayoutElementGetCurrentContext().needsVisualizeNode && !self.neverShouldVisualize) {
+    return [[ASLayoutSpecVisualizerNode alloc] initWithLayoutSpec:self];
+  } else {
+    return self;
+  }
+}
+
+- (void)recursivelySetShouldVisualize:(BOOL)visualize
```

Comment:
If there is a DEBUG type flag that works in the ASDK framework and is toggled by the default Xcode configuration for most apps (I'm not sure there is)...then we could wrap methods like this to be debug-only.  We can probably clean up the count check at the bottom, but it doesn't pose a risk to production code.